pkg/collect/logs.go

```diff
@@ -13,17 +14,15 @@ import (
 	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/client-go/kubernetes"
-	"k8s.io/client-go/rest"
 )
 
 type CollectLogs struct {
-	Collector    *troubleshootv1beta2.Logs
-	BundlePath   string
-	Namespace    string
-	ClientConfig *rest.Config
```

Comment:
ihmo we need to revert this one and create a follow up to discuss if we should pass the client or the config to create a new client for each collector. My 2 cents here is that we need create a new client for each collector for not bitten for the rate-limiter.

Anyway, the change need to be made horizontally ( same for all collectors ).

Reply:
At the moment the the logs collector is the only one doing creating its own client. The reason I removed it is to ensure the code is testable. This way we can pass in a client fake to create mocks.

Reply:
Hi @banjoh,

So, I think you just discover why all collector func receive both the client and the config:
-> Client : to help on the tests 
-> config : for the real code use the config to get a new client so